pkgs/development/compilers/openjdk/generic.nix

```diff
@@ -0,0 +1,664 @@
+  # still runs in parallel.
+  enableParallelBuilding = false;
+
+  env =
+    {
+      NIX_CFLAGS_COMPILE =
+        if atLeast17 then
+          "-Wno-error"
+        else if atLeast11 then
+          # Workaround for
+          # `cc1plus: error: '-Wformat-security' ignored without '-Wformat' [-Werror=format-security]`
+          # when building jtreg
+          "-Wformat"
+        else
+          lib.concatStringsSep " " (
+            [
+              # glibc 2.24 deprecated readdir_r so we need this
+              # See https://www.mail-archive.com/openembedded-devel@lists.openembedded.org/msg49006.html
+              "-Wno-error=deprecated-declarations"
+            ]
+            ++ lib.optionals stdenv.cc.isGNU [
+              # https://bugzilla.redhat.com/show_bug.cgi?id=1306558
+              # https://github.com/JetBrains/jdk8u/commit/eaa5e0711a43d64874111254d74893fa299d5716
+              "-fno-lifetime-dse"
+              "-fno-delete-null-pointer-checks"
+              "-std=gnu++98"
+              "-Wno-error"
+            ]
+          );
+
+      NIX_LDFLAGS = lib.concatStringsSep " " (
+        lib.optionals (!headless) [
+          "-lfontconfig"
+          "-lcups"
+          "-lXinerama"
+          "-lXrandr"
+          "-lmagic"
+        ]
+        ++ lib.optionals (!headless && enableGtk) [
+          (if atLeast11 then "-lgtk-3" else "-lgtk-x11-2.0")
+          "-lgio-2.0"
+        ]
+      );
+    }
+    // lib.optionalAttrs (!atLeast11) {
+      # OpenJDK 8 Hotspot cares about the host(!) version otherwise
+      DISABLE_HOTSPOT_OS_VERSION_CHECK = "ok";
+    };
+
+  ${if atLeast23 then "versionCheckProgram" else null} = "${placeholder "out"}/bin/java";
+
+  ${if !atLeast11 then "doCheck" else null} = false; # fails with "No rule to make target 'y'."
+
+  doInstallCheck = atLeast23;
+
+  ${if atLeast17 then "postPatch" else null} = ''
+    chmod +x configure
+    patchShebangs --build configure
+  '';
+
+  ${if !atLeast17 then "preConfigure" else null} =
+    ''
+      chmod +x configure
+      substituteInPlace configure --replace /bin/bash "${bash}/bin/bash"
+    ''
+    + lib.optionalString (!atLeast11) ''
+      substituteInPlace hotspot/make/linux/adlc_updater --replace /bin/sh "${stdenv.shell}"
+      substituteInPlace hotspot/make/linux/makefiles/dtrace.make --replace /usr/include/sys/sdt.h "/no-such-path"
+    '';
+
+  installPhase =
+    ''
+      mkdir -p $out/lib
+
+      mv build/*/images/${if atLeast11 then "jdk" else "j2sdk-image"} $out/lib/openjdk
+
+      # Remove some broken manpages.
+      rm -rf $out/lib/openjdk/man/ja*
+
+      # Mirror some stuff in top-level.
+      mkdir -p $out/share
+      ln -s $out/lib/openjdk/include $out/include
+      ln -s $out/lib/openjdk/man $out/share/man
+    ''
+    + lib.optionalString atLeast17 ''
+
+      # IDEs use the provided src.zip to navigate the Java codebase (https://github.com/NixOS/nixpkgs/pull/95081)
+    ''
```

Comment:
Why is this comment by itself >=17, when the actual code it refers to is >=11?

Reply:
Because it was changed between `11.nix` and `17.nix`, and this affects the derivation hash.